addons/website_mail_channel/controllers/main.py

```diff
@@ -175,8 +175,19 @@ def thread_discussion(self, group, message, mode='thread', date_begin=None, date
             base_domain = [('model', '=', 'mail.channel'), ('res_id', '=', group.id)]
         next_message = Message.search(base_domain + [('date', '<', message.date)], order="date DESC", limit=1) or None
         prev_message = Message.search(base_domain + [('date', '>', message.date)], order="date", limit=1) or None
+
+        # handle access to attachments for public / portal
+        if request.env.user.has_group('base.group_user'):
+            attachments = message.attachment_ids
+        else:
+            # portal / external people cannot access attachments -> grant access
+            # as message is ok, generate access tokens
+            attachments = message.sudo().attachment_ids
+            attachments.generate_access_token()
+
         values = {
             'message': message,
+            'attachments': attachments,
```

Comment:
Is that a new feature? Or fix out of scope of current task?
I don't get why we suddently add this here.